rust/extractor/src/main.rs

```diff
@@ -277,6 +277,16 @@ fn main() -> anyhow::Result<()> {
     } else {
         ResolvePaths::Yes
     };
+    let library_mode = if cfg.extract_dependencies_as_source {
+        SourceKind::Source
+    } else {
+        SourceKind::Library
+    };
+    let library_resolve_paths = if cfg.extract_dependencies_as_source {
+        resolve_paths
+    } else {
+        ResolvePaths::No
+    };
```

Comment:
```suggestion
    let (library_mode, library_resolve_paths) = if cfg.extract_dependencies_as_source {
        (SourceKind::Source, resolve_paths)
    } else {
        (SourceKind::Library, ResolvePaths::No)
    };
```
Avoids repeating the conditional.